File: uploaddemo/src/business/galary/java/com/jw/shotRecord/video/VideoItem.java
//
// Source code recreated from a .class file by IntelliJ IDEA
// (powered by Fernflower decompiler)
//

package com.jw.shotRecord.video;

import android.os.Parcel;
import android.os.Parcelable;

import java.io.Serializable;

public class VideoItem implements Serializable, Parcelable {
    public String name;
    public String path;
    public String thumbPath;
    public long size;
    public long duration;
    public String mimeType;
    public static final Creator<VideoItem> CREATOR = new Creator<VideoItem>() {
        public VideoItem createFromParcel(Parcel source) {
            return new VideoItem(source);
        }

        public VideoItem[] newArray(int size) {
            return new VideoItem[size];
        }
    };

    public boolean equals(Object o) {
        if (o instanceof VideoItem) {
            VideoItem item = (VideoItem) o;
            return this.path.equalsIgnoreCase(item.path);
        } else {
            return super.equals(o);
        }
    }

    public int describeContents() {
        return 0;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.name);
        dest.writeString(this.path);
        dest.writeLong(this.size);
        dest.writeString(this.mimeType);
        dest.writeLong(this.duration);
        dest.writeString(this.thumbPath);
    }

    public VideoItem() {
    }

    protected VideoItem(Parcel in) {
        this.name = in.readString();
        this.path = in.readString();
        this.size = in.readLong();
        this.mimeType = in.readString();
        this.duration = in.readLong();
        this.thumbPath = in.readString();
    }
}

File: library/src/main/java/com/jw/library/model/VideoItem.kt
package com.jw.library.model

import android.os.Parcel
import android.os.Parcelable

/**
 * 视频实体类
 * @property thumbPath String?  缩略图地址
 * @property duration Long  时长
 */
class VideoItem : BaseItem {
    var thumbPath: String? = null
    var duration: Long = 0

    constructor()

    constructor(path: String, thumbPath: String, duration: Long) : super(path) {
        this.thumbPath = thumbPath
        this.duration = duration
    }

    constructor(parcel: Parcel) : super(parcel) {
        duration = parcel.readLong()
        thumbPath = parcel.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        super.writeToParcel(parcel, flags)
        parcel.writeLong(duration)
        parcel.writeString(thumbPath)
    }

    companion object CREATOR : Parcelable.Creator<VideoItem> {
        override fun createFromParcel(parcel: Parcel): VideoItem {
            return VideoItem(parcel)
        }

        override fun newArray(size: Int): Array<VideoItem?> {
            return arrayOfNulls(size)
        }
    }
}
